### ISSUE TITLE
Standardise a fallback mechanism (summary property?) for unrecognised entries?

### ISSUE THREAD

**Issue by Zegnat** [microformats/h-entry#9]
As `h-entry` is getting used for more and more different types of posts (IndieWeb: [PostType](https://indieweb.org/Category:PostType), SWWG: [Post Type Discovery](https://www.w3.org/TR/post-type-discovery/)) and as a way to move those posts around ([Micropub](https://micropub.net/)) the amount of cases where a consumer doesn’t know what to do with the entry grows.

Having a form of human-friendly description available as fallback has come up several times over the years. When a consumer does not understand a combination of properties it may rely on having a plain text alternative instead.

Not only posts using draft properties or fully experimental properties are helped by this. Even posts that only use core properties such as a [like post](https://indieweb.org/like) can be hard for consumers to deal with due to a (possible) lack of `content`. As an example: [Loqi](https://indieweb.org/Loqi) has no special handling for likes and would use [fallback properties](https://chat.indieweb.org/dev/2016-08-10#t1470859728768000).

Implementers of microformats consumers could be helped by us specifically calling out a property as “good to use as fallback”. The `h-entry` vocabulary would be a prime page to do this. Example: @manton [wrote](https://chat.indieweb.org/dev/2018-06-10/1528643205828400) that he doesn’t want people scared by all the potential properties, I think being able to say “just make sure to implement the fallback mechanism” goes a long way there.

**TL;DR:** should we codify `summary` as a proper fallback that consumers can use when the entry contains other properties that aren’t understood? Are there special cases where this would not be optimal?

---

**Long read:** I tried to pull in some history, some current debates, and some iteration on fallback properties. This should give a pretty good overview of the thoughts behind the above TL;DR and this issue.

A [read post](https://indieweb.org/read) [implementation](https://gregorlove.com/2017/11/martijn-asked-if-i-had/) by @gRegorLove used the `content` property as an HTML fallback:

> The `h-cite` object also has a `content` property which has an `html` key. The value of that key is the HTML for the citation. This serves as a fallback in case the site does not process the micropub `h-cite` object directly.

In the latest iteration of the read posts, as posted through [indiebookclub](https://indiebookclub.biz/), this has been dropped entirely in favour of the `summary` property. See microdotblog/issues#92 for details on the [microformats2 JSON](http://microformats.org/wiki/microformats2-json) used for Micropub.

On that same issue, @aaronpk writes in [to elaborate](https://github.com/microdotblog/issues/issues/92#issuecomment-396052115) on why `summary` is a better fallback than `content`. Quoting in full:

> The `summary` property is intended to be used for fallback for when the server doesn't recognize the other properties of the request. Most Micropub servers understand `content` and `photo`, but this is a great example of an interesting content type that is likely not supported by most servers yet. In this case, `summary` is the correct place for this fallback content. The `content` property would be misleading since normally the `content` property contains user-entered text. For example, if the user enters a note about why they wanted to read this, that could be included in the `content` property.

This point was also made [in the IndieWeb chat](https://chat.indieweb.org/dev/2018-06-10#t1528643315766200):

* > the functional difference between summary and `content is demonstrated by a photo post that has no caption. it would be misleading to have the client generate a summary of the photo and include that in content, since then the server would have no way of knowing whether the content was something the user entered or not

Using the `summary` property as fallback has been brought up several times. Including by @tantek in [discussions about parsing comments](https://chat.indieweb.org/social/2016-03-16#t1458152746367000):

* > ... So when people started doing lieks via webmention, by including a summary in your like that says 'so and so likes this'
* >  ... if you understand likes you ignore the summary and just increment your counter or whatever

The Micropub standard also notes the `summary` property in its section on [unrecognised properties](https://www.w3.org/TR/micropub/#unrecognized-properties):

> For example, a client may create a post that contains a weight measurement, with an `h-measure` value for the weight property, and a plaintext summary of the post in the `summary` property. Servers that don't recognize the `weight` property will simply ignore it, and will create the post as a plaintext post with the summary instead.

The Micropub standard also calls out that unrecognised properties may just be ignored:

> If the request includes properties that the server does not recognize, it *MUST* ignore unrecognized properties and create the post with the values that are recognized.

This may lead to unexpected consumer behaviour and is the only pain-point I can think of, as I [wrote in the chat](https://chat.indieweb.org/dev/2018-06-10#t1528645402155000):

* > I think it is hard to figure out exactly when to fall back to summary. In the photo/content/summary case, you want to ignore content and use summary if you do not support photo. But in the photo/content/summary/location case, if you do support photo but not location, you probably only want to ignore location rather than fallback to only summary :(

To stick with the [photo post](https://indieweb.org/photo):

* if an entry contains `photo`, `content`, and `summary` properties and the consumer does not understand `photo` then `summary` should be used by itself. Using `content` when this relies on the photograph is unwanted.
  
* if an entry contains `photo`, `content`, `summary`, and `location` properties and the consumer does not understand `location` then… what? Ignoring `location` and using the `photo` and `content` properties for a bare photo post seems to make sense in some context.

This makes it hard to just say “fallback to `summary` when the combination of other properties is unknown”.

Also note: non of the discussions I was able to find talk about how this overlaps with the use of `summary` for `h-entry` objects within `h-feed` objects. Could advertising `summary` as universal fallback have a negative impact there?

**Comment by Zegnat** [comment]
Whelp, that got a bit long! But I hope it captured all the different thoughts and angles, centralising the entire discussion to this one issue 😄

**Comment by kevinmarks** [comment]
This makes sense, but what about name? That should be used to show the
sense, with summary as a more detailed version.

By removing implied name, have we made the ability to use name as a minimal
version of any post format harder?

**Comment by Zegnat** [comment]
> By removing implied name, have we made the ability to use name as a minimal version of any post format harder?

Probably yes. Were consumers actually using `name` “as a minimal version of any” unrecognised post? I’m not sure. Most consumers that weighed in on the implied name issue were having problems because `name` was always present and it was hard to know when it was actually a usable title.

This is why I also brought up `summary` in the context of feeds: if summaries in that context are being used in a way that makes them hard to be used as fallback we might need a different property. Else we run into the issue we tried to solve around `name` where consumers have a hard time distinguishing fallbacks from e.g. small teasers used in feed context.

I think this risk might be minimal. As a properly written summary for a feed will often work well as a stand-alone fallback for the entry.

**Comment by kevinmarks** [comment]
There is demand for previewing entries in general, and I think that the name, summary, featured image combination is the pattern that is being converged on - see discussion here: https://github.com/ampproject/amphtml/issues/657

Is the distinction between fallback and teaser significant if there is a link to the full version, or is this like the problem with permalinks to notes where clicking through gives no more information?

Twitter moments have the reverse problem, where the presentation of a tweet there is often less coherent than inline, because quoted tweets are not expanded; tweets have grown richer by showing upstream and downstream threads, so clicking through can still have benefit. 

Encouraging non-duplicative use of summary is a good idea.
